docs/proposals/timeseries-partitioning-in-compactor.md

```diff
@@ -124,34 +124,27 @@ T1 - Partition 1-2 was created with hash % 2 == 0, and in order to avoid having
 
 ### Compaction Workflow
 
-1. Cortex Compactor initializes grouper and planner. Then, call Thanos Compactor to compact blocks.
-2. Thanos Compactor retrieves block meta and call Grouper to group blocks for compaction.
+1. Compactor initializes Grouper and Planner.
+2. Compactor retrieves block's meta.json and call Grouper to group blocks for compaction.
 3. Grouper generates compaction plans:
    1. Grouper groups source blocks into unpartitioned groups.
    2. For each unpartitioned group:
       1. Calculates partition number. Partition number indicates how many partitions one unpartitioned group would be partitioned into based on the total size of indices and number of time series from each source blocks in the unpartitioned group.
       2. Assign source blocks into each partition with partition ID (value is in range from 0 to Partition Number - 1). Note that one source block could be used in multiple partitions. So multiple partition ID could be assigned to same source block. Check the examples in [Compaction Partitioning Examples](#compaction-partitioning-examples)
-      3. Gather all partition IDs got assigned to each source blocks.
-      4. For each source block, save compaction plan file into block-storage. Compaction plan file would look like:
-         ```
-         {"partitionIDs": [0, 1, ...]}
-         ```
-4. Grouper returns compaction plans to Thanos Compactor.
-5. Thanos Compactor iterates over each compaction plan (partitioned group). For each iteration, calls Planner to make sure the group is ready for compaction.
+      3. Generates compaction plan that indicates which partition ID each blocks got assigned.
+4. Grouper returns compaction plans to Compactor.
```

Comment:
Probably better to update the wording. Grouper should generate `compaction group` rather than `compaction plans`.
Planning should be done on the planner side.

Reply:
I want to stick with `compaction plan`. I will discuss with the team to see if it makes sense we put partitioning logic inside planner. Because we are thinking store partitioned compaction plan in a centralized location in block storage so that any compactor can pick up the plan once generated.

Reply:
Let's update the proposal then. But the centralized location you mean is the object storage?